runtime/server/column_api_test.go

```diff
@@ -52,3 +49,137 @@ func TestServer_GetTopK(t *testing.T) {
+	}
+	res, err := server.EstimateSmallestTimeGrain(context.Background(), &api.EstimateSmallestTimeGrainRequest{InstanceId: instanceId, TableName: "test", ColumnName: "times"})
+	if err != nil {
+		t.Error(err)
+	}
+	require.NotNil(t, res)
+	require.Equal(t, "DAYS", res.TimeGrain.String())
+}
+
+func TestServer_GetNumericHistogram(t *testing.T) {
+	server, instanceId, err := getTestServerWithData(t)
+
+	res, err := server.GetNumericHistogram(context.Background(), &api.NumericHistogramRequest{InstanceId: instanceId, TableName: "test", ColumnName: "val"})
+	if err != nil {
+		t.Error(err)
+	}
+	require.NotNil(t, res)
+	require.Equal(t, 3, len(res.NumericHistogramBins.Bins))
+	require.Equal(t, int64(0), res.NumericHistogramBins.Bins[0].Bucket)
+	require.Equal(t, 1.0, res.NumericHistogramBins.Bins[0].Low)
+	require.Equal(t, 2.333333333333333, res.NumericHistogramBins.Bins[0].High)
+	require.Equal(t, int64(2), res.NumericHistogramBins.Bins[0].Count)
+}
+
+func TestServer_GetCategoricalHistogram(t *testing.T) {
+	server, instanceId, err := getTestServerWithData(t)
+
+	res, err := server.GetRugHistogram(context.Background(), &api.RugHistogramRequest{InstanceId: instanceId, TableName: "test", ColumnName: "val"})
+	if err != nil {
+		t.Error(err)
+	}
+
+	require.NotNil(t, res)
+	require.Equal(t, 3, len(res.NumericOutliers.Outliers))
+	require.Equal(t, int64(0), res.NumericOutliers.Outliers[0].Bucket)
+	require.Equal(t, 1.0, res.NumericOutliers.Outliers[0].Low)
+	require.Equal(t, 1.008, res.NumericOutliers.Outliers[0].High)
+	require.Equal(t, true, res.NumericOutliers.Outliers[0].Present)
+
+	// works only with numeric columns
+	res, err = server.GetRugHistogram(context.Background(), &api.RugHistogramRequest{InstanceId: instanceId, TableName: "test", ColumnName: "times"})
+	require.ErrorContains(t, err, "Conversion Error: Unimplemented type for cast (TIMESTAMP -> DOUBLE)")
+}
+
+func TestServer_GetTimeRangeSummary(t *testing.T) {
+	server, instanceId, err := getTestServerWithData(t)
+
+	// Get Time Range Summary works with timestamp columns
+	res, err := server.GetTimeRangeSummary(context.Background(), &api.TimeRangeSummaryRequest{InstanceId: instanceId, TableName: "test", ColumnName: "times"})
+	if err != nil {
+		t.Error(err)
+	}
+	require.NotNil(t, res)
+	require.Equal(t, "2022-11-01 00:00:00 +0000 UTC", res.Min)
+	require.Equal(t, "2022-11-03 00:00:00 +0000 UTC", res.Max)
+	require.Equal(t, int32(0), res.Interval.Months)
+	require.Equal(t, int32(2), res.Interval.Days)
+	require.Equal(t, int64(0), res.Interval.Micros)
+}
+
+func TestServer_GetCardinalityOfColumn(t *testing.T) {
+	server, instanceId, err := getTestServerWithData(t)
+
+	// Get Cardinality of Column works with all columns
+	res, err := server.GetCardinalityOfColumn(context.Background(), &api.CardinalityOfColumnRequest{InstanceId: instanceId, TableName: "test", ColumnName: "val"})
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
nit: use `require.NoError(t, err)` for all these

Reply:
changed